vault/token_store.go

```diff
@@ -1325,32 +1339,58 @@ func (ts *TokenStore) handleUpdateLookupAccessor(req *logical.Request, data *fra
 // the token associated with the accessor
 func (ts *TokenStore) handleUpdateRevokeAccessor(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
 	var urlaccessor bool
-	accessor := data.Get("accessor").(string)
-	if accessor == "" {
-		accessor = data.Get("urlaccessor").(string)
-		if accessor == "" {
+	accessors := data.Get("accessor").([]string)
+
+	if len(accessors) == 0 {
+		accessors = data.Get("urlaccessor").([]string)
+		if len(accessors) == 0 {
 			return nil, &logical.StatusBadRequest{Err: "missing accessor"}
 		}
 		urlaccessor = true
 	}
 
-	aEntry, err := ts.lookupByAccessor(accessor, true)
-	if err != nil {
-		return nil, err
+	errs := make([]error, len(accessors))
+	tokens := make([]string, len(accessors))
+
+	for idx, accessor := range accessors {
+		aEntry, err := ts.lookupByAccessor(accessor, true)
+		if err != nil {
+			errs[idx] = err
+			tokens[idx] = ""
+		}
+
+		tokens[idx] = aEntry.TokenID
 	}
 
-	// Revoke the token and its children
-	if err := ts.RevokeTree(aEntry.TokenID); err != nil {
-		return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
+	revokeErrors := ts.RevokeTrees(tokens)
+
+	response := &logical.Response{}
+	failedRevokes := make([]map[string]string, 0, len(revokeErrors))
+
+	for idx, revokeError := range revokeErrors {
+		if errs[idx] == nil {
+			errs[idx] = revokeError
+		}
+
+		if errs[idx] != nil {
+			failedRevokes = append(failedRevokes, map[string]string{
```

Comment:
When something returns multiple errors, we normally use [hashicorp/go-multierror](https://github.com/hashicorp/go-multierror) and I think it would work in this case.  I think it would remove a lot of the backwards compatibility logic since it would just contain one error in the case of a single accessor.